frontend/src/app/auth.service.spec.ts

```diff
@@ -1,16 +1,58 @@
-import {TestBed} from '@angular/core/testing'
-
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing'
 import {AuthService} from './auth.service'
+import {RouterTestingModule} from '@angular/router/testing'
+import {TestBed, getTestBed, fakeAsync, tick} from '@angular/core/testing'
+
+const testUser: Scrum.User = {
+  displayName: 'User Display Name',
+  pk: 1,
+  email: 'user@user.com',
+  is_PM: false,
+}
+
+let httpMock: HttpTestingController
 
 describe('AuthService', () => {
   let service: AuthService
+  let injector: TestBed
 
   beforeEach(() => {
-    TestBed.configureTestingModule({})
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, RouterTestingModule],
+    })
     service = TestBed.inject(AuthService)
+    injector = getTestBed()
+    httpMock = injector.get(HttpTestingController)
   })
 
   it('should be created', () => {
     expect(service).toBeTruthy()
   })
+
+  it('should get properties of authenticated user', async () => {
+    service.user = testUser
+    Object.entries(testUser).forEach(async ([key, val]) => {
+      const queriedVal = await service.queryUser(key).toPromise()
+      expect(queriedVal).toEqual(testUser[key])
+    })
+  })
+  it('should get the entire authenticated user', async () => {
+    service.user = testUser
+    const queriedUser = await service.getUserInfo().toPromise()
+    expect(queriedUser).toBe(testUser)
+  })
+
+  it('should return the authed user when creds are right', () => {
+    ;(service as any)
```

Comment:
this as any business is because submitCreds is a private method on this service.  theres apparently no universally great way to call it in the context of a test like this, so this basically tells typescript to just Deal With It